apollo-router/src/services/connector_service.rs

```diff
@@ -167,7 +168,24 @@ async fn execute(
                 http_request: req,
                 context,
             };
-            let res = client.oneshot(req).await?;
+            let res = client.oneshot(req).await.map_err(|e| {
+                match e.downcast::<FetchError>() {
+                    // Replace the internal subgraph name with the connector label
+                    Ok(inner) => match *inner {
+                        FetchError::SubrequestHttpError {
```

Comment:
a few different FetchError variants include service name: https://github.com/apollographql/router/blob/dev/apollo-router/src/error.rs#L66,75 ... patch response, ws, and batching are unlikely to happen but should we check for them anyway?

Reply:
The fetch service [forces everything into `SubrequestHttpError`](https://github.com/apollographql/router/blob/next/apollo-router/src/services/fetch.rs#L107-L128). However, I missed that it's setting the `reason` using the `Display` implementation of the original error, and since that's derived, it will include the `service` field. I'll need to fix.

Reply:
Actually, looking at it more, those errors are never returned except through the subgraph service, so they would never happen for connector service. The only error that can come out of the connector service is `SubrequestHttpError`, either from the HTTP client service (which this PR handles) or directly [emitted by the fetch service](https://github.com/apollographql/router/blob/0b67964fccb367f20b89ee64c66cd62a9b9e9509/apollo-router/src/plugins/connectors/handle_responses.rs#L159), which already returned the connector label.